### PR TITLE
static-class-names-order rule

### PR DESCRIPTION
**Pull request by mchmurski-rms**
https://github.com/vuejs/eslint-plugin-vue/issues/885

### CODE REVIEW

File: docs/rules/README.md [hunk 1]
```diff
@@ -161,6 +162,8 @@ For example:
 | [vue/space-infix-ops](./space-infix-ops.md) | require spacing around infix operators | :wrench: |
 | [vue/space-unary-ops](./space-unary-ops.md) | enforce consistent spacing before or after unary operators | :wrench: |
 | [vue/v-on-function-call](./v-on-function-call.md) | enforce or forbid parentheses after method calls without arguments in `v-on` directives | :wrench: |
+| [vue/v-slot-style](./v-slot-style.md) | enforce `v-slot` directive style | :wrench: |
+| [vue/valid-v-slot](./valid-v-slot.md) | enforce valid `v-slot` directives |  |
```
Comment (mchmurski-rms) [inline docs/rules/README.md]: These got autogenerated while doing `npm start update`

File: lib/rules/class-order.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+      category: undefined
+    },
+    fixable: 'code',
+    schema: []
+  },
+  create: context => {
+    return utils.defineTemplateBodyVisitor(context, {
+      "VAttribute[directive=false][key.name='class']" (node) {
+        const classList = node.value.value
+        const classListSorted = classList.split(' ').sort().join(' ')
```
Comment (ota-meshi) [inline lib/rules/class-order.js]: ASCII whitespace can be used to split classNames, Could you change it to support it?

https://infra.spec.whatwg.org/#ascii-whitespace

File: tests/lib/rules/class-order.js [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+tester.run('class-order', rule, {
+
+  valid: [
+    {
+      filename: 'test.vue',
+      code: '<template><div></div></template>'
+    },
+    {
+      filename: 'test.vue',
+      code: '<template><div class="a b"></div></template>'
```
Comment (ota-meshi) [inline tests/lib/rules/class-order.js]: Could you add testcases with multiple spaces?

```js
'<template><div class="a        b"></div></template>'
```

File: lib/index.js [hunk 4]
```diff
@@ -14,6 +14,7 @@ module.exports = {
     'block-spacing': require('./rules/block-spacing'),
     'brace-style': require('./rules/brace-style'),
     'camelcase': require('./rules/camelcase'),
+    'class-order': require('./rules/class-order'),
```
Comment (ota-meshi) [inline lib/index.js]: I think the name of the rule is misleading.
People may misunderstand that it targets `:class` or JS `class`.

Could you change the rule name?

For example, what do you think about changing the name of the rule to "static-class-names-order"?

File: docs/rules/class-order.md [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+pageClass: rule-details
+sidebarDepth: 0
+title: vue/class-order
+description: enforce classnames order
+---
+# vue/class-order
+> enforce classnames order
+
+- :wrench: The `--fix` option on the [command line](https://eslint.org/docs/user-guide/command-line-interface#fixing-problems) can automatically fix some of the problems reported by this rule.
+
```
Comment (ota-meshi) [inline docs/rules/class-order.md]: Could you add some sample code?

https://github.com/vuejs/eslint-plugin-vue/blob/master/docs/rules/attributes-order.md

File: lib/rules/class-order.js (hunk 2 above)
Comment (mchmurski-rms) [reply lib/rules/class-order.js]: I've used regex with `\s` this time

File: tests/lib/rules/class-order.js (hunk 3 above)
Comment (mchmurski-rms) [reply tests/lib/rules/class-order.js]: done

File: lib/index.js (hunk 4 above)
Comment (mchmurski-rms) [reply lib/index.js]: done

File: docs/rules/class-order.md (hunk 5 above)
Comment (mchmurski-rms) [reply docs/rules/class-order.md]: done

**Comment by mchmurski-rms** [comment]
@ota-meshi thanks for the review, this should be ready for another round

**Comment by mchmurski-rms** [comment]
@ota-meshi thanks, this should be ready for review

File: docs/rules/static-class-names-order.md [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+
+<eslint-code-block fix :rules="{'vue/static-class-names-order': ['error']}">
+
+```vue
+<template>
+  <!-- ✓ GOOD -->
+  <div class="a b"></div>
+
+  <!-- ✗ BAD -->
+  <div class="b a></div>
```
[the remark this answers is not in the document]
Comment (mchmurski-rms) [reply docs/rules/static-class-names-order.md]: my bad, a typo, done!

**Pull request merged**